lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -213,6 +213,47 @@ impl LocalShard {
 
         Ok(records)
     }
+
+    /// Ensures that the `available disk space` is at least >= `wal_capacity_mb`.
+    ///
+    /// This function checks the available disk space on the file system where the specified path is located.
+    /// It verifies that there is at least enough free space to meet the WAL buffer size configured for the collection.
+    /// Introducing this check helps prevent service disruptions from unexpected disk space exhaustion.
+    ///
+    /// # Returns
+    /// A result indicating success (`Ok(())`) if there is sufficient disk space,
+    /// or a `CollectionError::service_error` if the disk space is insufficient or cannot be retrieved.
+    ///
+    /// # Errors
+    /// This function returns an error if:
+    /// - The disk space retrieval fails, detailing the failure reason.
+    /// - The available space is less than the configured WAL buffer size, specifying both the available and required space.
+    async fn ensure_sufficient_disk_space(&self) -> CollectionResult<()> {
+        let disk_free_space_bytes = fs2::available_space(self.path.as_path()).map_err(|err| {
```

Comment:
calling sync `available_space` in async context might be not so good performance-wise